Sources/Commands/SwiftTestTool.swift

```diff
@@ -205,6 +211,11 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
             let testSuites = try getTestSuites(path: testPath)
             let tests = testSuites.filteredTests(specifier: options.testCaseSpecifier)
 
+            // if user called --num-workers without the --parallel option, throw an error
+            if (options.numberOfWorkers>0) && !options.shouldRunInParallel {
+                throw TestError.invalidUsage(argument: "--num-workers")
```

Comment:
We should use emit a diagnostic here. Take a look at `MutuallyExclusiveArgumentsDiagnostic` for example. The output should be something like:

```
--num-workers should be used with the --parallel
``